src/uu/mv/src/mv.rs

```diff
@@ -751,3 +845,370 @@ fn is_empty_dir(path: &Path) -> bool {
+                                    format!("error writing {}: {}", dir_entry_to.quote(), error)
+                                }
+                                _ => {
+                                    format!("{:?}", err)
+                                }
+                            };
+                            error_occurred = true;
+                            show_error!("{}", err_msg);
+                            continue;
+                        }
+                    }
+                    if let Some(vc) = vebose_context.as_ref() {
+                        vc.print_copy_file(&dir_entry_path, &dir_entry_to, false);
+                    }
+                }
+                moved_entries.push((dir_entry_path, file_type));
+            }
+            Err(err) => {
+                let err_msg = match (err.io_error(), err.path()) {
+                    (Some(io_error), Some(path)) => {
+                        format!("cannot access {}: {io_error}", path.quote())
+                    }
+                    _ => err.to_string(),
+                };
+                show_error!("{err_msg}");
+                error_occurred = true;
+            }
+        }
+    }
+    if !error_occurred {
+        while !moved_entries.is_empty() {
+            let (moved_path, file_type) = moved_entries.pop().unwrap();
+            let res = if file_type.is_dir() {
+                remove(&moved_path)
+            } else {
+                file::remove(&moved_path)
+            };
+            if let Err(err) = res {
+                error_occurred = true;
+                show_error!("cannot remove {}: {}", moved_path.quote(), err);
+            } else {
+                if let Some(vc) = vebose_context.as_ref() {
+                    if file_type.is_dir() {
+                        vc.remove_folder(&moved_path);
+                    } else {
+                        vc.remove_file(&moved_path);
+                    }
+                }
+            }
+        }
+    }
+    if error_occurred {
+        return Err(MvError::NotAllFilesMoved);
+    }
+    Ok(result)
+}
+
+/// Copies a file from one path to another, updating the progress bar if provided.
+fn copy_file(
+    from: &Path,
+    to: &Path,
+    progress_bar: Option<&ProgressBar>,
+    progress_bar_start_val: u64,
+) -> FsXResult<u64> {
+    let copy_options: CopyOptions = CopyOptions {
+        // We are overwriting here based on the assumption that the update and
+        // override options are handled by a parent function call.
+        overwrite: true,
+        ..Default::default()
+    };
+    let progress_handler = if let Some(progress_bar) = progress_bar {
+        let display_file_name = from
+            .file_name()
+            .and_then(|file_name| file_name.to_str())
+            .map(|file_name| file_name.to_string())
+            .unwrap_or_default();
+        let _progress_handler = |info: file::TransitProcess| {
```

Comment:
A leading underscore indicates that the variable isn't actually used, and only exists due to the syntax.

This is misleading here, so please remove the leading underscore.